modules/standard/IO.chpl

```diff
@@ -7162,36 +7162,36 @@ iter fileReader.lines(
     const f = chpl_fileFromReaderOrWriter(this),
           myStart = qio_channel_start_offset_unlocked(this._channel_internal),
           myEnd = qio_channel_end_offset_unlocked(this._channel_internal),
-          myBounds = myStart..<min(myEnd, try! f.size),
+          myBounds = myStart..min(myEnd, try! f.size),
           fpath = try! f.path;
 
     try {
       // try to break the file into chunks and read the lines in parallel
       // can fail if the file is too small to break into 'targetLocales.size' chunks
       const byteOffsets = findFileChunks(f, targetLocales.size, myBounds);
-
       coforall lid in 0..<targetLocales.size do on targetLocales[tld.orderToIndex(lid)] {
-        const locBounds = byteOffsets[lid]..byteOffsets[lid+1],
-              locFile = try! open(fpath, ioMode.r),
-              nTasks = if dataParTasksPerLocale>0 then dataParTasksPerLocale
-                                                  else here.maxTaskPar;
-
-        try {
-          // try to break this locale's chunk into 'nTasks' chunks and read the lines in parallel
-          const locByteOffsets = findFileChunks(locFile, nTasks, locBounds);
-          coforall tid in 0..<nTasks {
-            const taskBounds = locByteOffsets[tid]..<locByteOffsets[tid+1],
-                  r = try! locFile.reader(region=taskBounds);
-
-            if taskBounds.size > 0 {
-              var line: t;
-              while (try! r.readLine(line, stripNewline=stripNewline)) do
-                yield line;
+        const locBounds = byteOffsets[lid]..byteOffsets[lid+1];
+        if locBounds.size > 1 {
```

Comment:
Is this saying, if the number of bytes between this one and the next one is > 1, then proceed to read something? If it's == 1, could there be an empty line that should be read? I suspect there is some logic here that you know that should be reflected in a comment.

Reply:
Yeah, it's basically saying that if there is at least one byte to read in this locale's region then try to read it, otherwise skip it. I added a comment to explain that.

Reply:
Thanks, it addresses it for me. (I was also thinking it was a range like `x..<y` but it is `x..y` so `> 1` covers everything beyond `x..x`.